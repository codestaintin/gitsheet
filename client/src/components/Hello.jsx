import React from 'react';

const Hello = () => (
  <h1>This is where I am coming</h1>
);

export default Hello;