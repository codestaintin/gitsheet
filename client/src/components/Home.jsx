import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import Loader from 'react-loader-spinner';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import Search from './common/Search';
import Card from './common/Card';
import loadCheatsAction from '../actions/gitCheat/gitCheatActions';

/**
 * @export
 *
 * @class Home
 *
 * @extends {React.Component}
 */
export class Home extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      searchTerm: ''
    };
    this.handleSearch = this.handleSearch.bind(this);
  }

  componentDidMount() {
    this.props.loadCheatsAction();
  }

  handleSearch(event) {
    this.setState({ [event.target.name]: event.target.value });
  }

  render() {
    const { categories, isLoading } = this.props;
    return (
      <Fragment>
        &nbsp;
        <Search handleSearch={this.handleSearch} />
        &nbsp;
        <div className="container">
          {isLoading
        && (
          <div className="d-flex justify-content-center">
            <Loader type="ThreeDots" color="#3D4E81" className="text-center" />
          </div>
        )
          }
          <div className="card-columns">
            {
              categories
              && categories
                .filter(category => category.name.toLowerCase()
                  .includes(this.state.searchTerm.toLowerCase()))
                .map(category => {
                  return (
                    <Card
                      key={category._id}
                      {...category}
                    />
                  );
                })
            }
            {
              isLoading && (categories && !categories.length > 0)
              && (
                <h1 className="text-center">
                  Add new git cheats
                </h1>
              )
            }
          </div>
        </div>
      </Fragment>
    );
  }
}

Home.propTypes = {
  loadCheatsAction: PropTypes.func
};

const mapStateToProps = (
  { gitCheatReducer: { gitCheats: { categories }, isLoading } }) => ({
  categories,
  isLoading
});

const mapDispatchToProps = dispatch => bindActionCreators(
  { loadCheatsAction },
  dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(Home);