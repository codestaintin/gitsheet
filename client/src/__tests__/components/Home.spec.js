import React from 'react';
import { shallow } from 'enzyme';
import { Home } from '../../components/Home';

describe('<Home>', () => {
  const props = {
    loadCheatsAction: jest.fn(() => {})
  };
  const tree = shallow(<Home {...props} />);
  it('should render the Home component successfully', () => {
    expect(tree).toMatchSnapshot();
  });
  it('should simulate componentDidMount', () => {
    const spy = jest.spyOn(tree.instance(), 'componentDidMount');
    tree.instance().componentDidMount();
    expect(spy).toHaveBeenCalled();
  });
  it('should simulate handleSearch', () => {
    const spy = jest.spyOn(tree.instance(), 'handleSearch');
    tree.instance().handleSearch({
      target: {
        name: ''
      }
    });
    expect(spy).toHaveBeenCalled();
  });
  it('should simulate onSetSidebarOpen', () => {
    const spy = jest.spyOn(tree.instance(), 'onSetSidebarOpen');
    tree.instance().onSetSidebarOpen();
    expect(spy).toHaveBeenCalled();
  });
});