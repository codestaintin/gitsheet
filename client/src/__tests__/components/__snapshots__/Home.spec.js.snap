// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home> should render the Home component successfully 1`] = `
<Fragment>
   
  <Search
    handleSearch={[Function]}
  />
   
  <div
    className="container"
  >
    <div
      className="card-columns"
    />
  </div>
</Fragment>
`;
